src/luxonis_ml/scheme/v1/config_building_blocks/base_models/head.py

```diff
@@ -0,0 +1,70 @@
+from pydantic import BaseModel
+from .custom_base_model import CustomBaseModel
+from typing import Optional, Dict, Any, List, Tuple, Literal, Union, Annotated
+from ..enums import *
+from abc import ABC
+
+class HeadMetadataClassification(BaseModel):
+    """
+    Metadata for classification head. The following arguments are accepted:
+        - classes: Array of object class names recognized by the model;
+        - n_classes: Number of object classes recognized by the model;
+    """
+    classes: List[str]
+    n_classes: int
```

Comment:
If we also add is_softmax field to classification, then the fields of HeadMetadataClassification and HeadMetadataSegmentation are exactly the same, and we will have problems differentiating between them (line 65). I would thus leave it as it is for now.

Reply:
I think this is something that would be good to have solved. Segmentation and classification are indeed virtually the same task, so they should have the same fields. It would be confusing if only one had `is_softmax`.
 
If the same fields would cause issues with ambiguity, then we should find a different way how to disambigous between the different heads.

Reply:
Can we add some static naming to the heads metadata which is always present? The other option is to keep the family and then metadata must respect that family.

Reply:
As we expect the same metadata for both classification and segmentation, and this is merely a metadata class, we could also unify the two classes into one, e.g. HeadMetadataClassificationSegmentation class. Or is this too clumsy?

Reply:
I would still keep them separate for now.

Reply:
I'd also keep it separate. Even though the outputs are of the same "type", the semantics of the outputs are very different. I think adding a static name or some `task_type` field would be better.

Reply:
I'm adding an additional field called `label_type` to keep it consistent with Connor's data refactor. This solves the ambiguity.

Reply:
Where is this added?

Reply:
commit 850828754af36500aefca9f879b7c4ee85cfc606